src/pseudopeople/interface.py

```diff
@@ -193,31 +199,42 @@ def generate_american_community_survey(
     :param config: An optional override to the default configuration. Can be a path
         to a configuration YAML file or a dictionary.
     :param year: The survey date year (format YYYY) to include in the dataset. Will
-        return an empty pd.DataFrame if there are no data with this year. If None is
+        raise a ValueError if there is no data for this year. If None is
         provided, data from all years are included in the dataset.
     :param state: The state string to include in the dataset. Either full name or
-        abbreviation (e.g., "Ohio" or "OH"). Will return an empty pd.DataFrame if there are no
-        data pertaining to this state. If None is provided, data from all locations are
+        abbreviation (e.g., "Ohio" or "OH"). Will raise a ValueError if there is no
+        data for this state. If None is provided, data from all locations are
         included in the dataset.
     :param verbose: Log with verbosity if True.
     :return: A pd.DataFrame of simulated ACS data.
     :raises ConfigurationError: An incorrect config is provided.
     :raises DataSourceError: An incorrect pseudopeople input data source is provided.
     """
     user_filters = []
-    if year:
-        user_filters.extend(
-            [
-                (DATASETS.acs.date_column_name, ">=", pd.Timestamp(f"{year}-01-01")),
-                (DATASETS.acs.date_column_name, "<=", pd.Timestamp(f"{year}-12-31")),
-            ]
-        )
+    if year is not None:
```

Comment:
Why add `is not None`?

Reply:
This was added because we had if year -> append user_filters even if year wasn't a valid year. I think with the error handling now with the try except this is not necessary.

Reply:
I'm pretty sure it's still necessary, because if the user passes `year=0` and this were `if year` instead of `if year is not None`, then `user_filters` would still be an empty list when passed to `_generate_dataset`, which would presumably then load data for all years as if `year=None` had been passed, when instead an error should have been raised because `0` is not a valid year.